### PR TITLE
Docs: Add documentation on writing and running tests

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Docs. Definitely could use proof-reading.

## How was this tested?
👀

### CODE REVIEW

File: docs/tests.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## Defining World Tests
+
+In order to run tests from your world, you will need to create a `test` package within your world package. This can be
+done by creating a `test` directory with a file named `__init__.py` inside it inside your world. By convention, a base
+for your world tests can be created in this file that you can then import into other modules.
+
+### WorldTestBase
+
+In order to test basic functionality of varying options, as well as to test specific edge cases or that certain
+interactions in the world interact as expected, you will want to use the [worldTestBase](/test/bases.py). This class
```
Comment (black-sliver) [inline docs/tests.md:20]:
```suggestion
interactions in the world interact as expected, you will want to use the [WorldTestBase](/test/bases.py). This class
```
?

**Review by black-sliver — COMMENTED** [review]
1. I think there should be a sentence explaining when not to use WorldTestBase (i.e. https://github.com/ArchipelagoMW/Archipelago/blob/main/test/bases.py#L14 or a "raw" inittest.testcase)
2. Suggestion below

**Comment by ReverM** [comment]
I read it and after discussion with treble was able to undesrtand testing. I would suggest maybe adding the intended behaviour of the code snippet but other than that it was clear

**Comment by black-sliver** [comment]
So, this is a bit concerning. The doc should be written in a way where you understand it without further discussion with treble :sweat_smile:

**Review by black-sliver — APPROVED** [review]
The "further discussion" was about something deemed out of scope for this doc, so it should be fine.
